src/pyodide/internal/workers.py

```diff
@@ -117,15 +139,25 @@ def __init__(
         )
         super().__init__(js_resp.url, js_resp)
 
+    @classmethod
+    def from_response(cls, body: Body, response: "Response") -> "Response":
```

Comment:
`from_response` is a bit of an odd name. Is there an existing api like this? I would think to make it a method called `replace_body`. e.g,
```
def replace_body(self, new_body: Body) -> "Response":
    """
    Returns a new Response object with the same headers but an updated body
    """"
```